File: app/src/main/java/com/privat/contacts/domain/UsersRepository.java
package com.privat.contacts.domain;

import com.privat.contacts.domain.model.UserDomain;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;

public interface UsersRepository {
    Observable<List<UserDomain>> networkUsers();
    Observable<List<UserDomain>> favoriteUsers();
    Completable fetchNewUser();
    Completable changeUserFavorite(int userId);
    Completable saveUser(int userId);
    Completable clearTempData();
    Observable<UserDomain> getUserById(int userId);
}

File: app/src/main/java/com/privat/contacts/domain/UsersRepository.kt
package com.privat.contacts.domain

import com.privat.contacts.domain.model.UserDomain
import io.reactivex.Completable
import io.reactivex.Observable

interface UsersRepository {
    fun networkUsers(): Observable<List<UserDomain>>
    fun favoriteUsers(): Observable<List<UserDomain>>
    fun fetchNewUser(): Completable
    fun changeUserFavorite(userId: Int): Completable
    fun saveUser(userId: Int): Completable
    fun clearTempData(): Completable
    fun getUserById(userId: Int): Observable<UserDomain>
}
